### PR TITLE
flashprog: 1.2 -> 1.3

### PR DESCRIPTION
**Pull request by felixsinger**
Release notes: https://flashprog.org/wiki/Flashprog/v1.3

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by felixsinger** [comment]
@NixOS/nixpkgs-merge-bot merge

**Review by funkeleinhorn — APPROVED** [review]
LGTM

**Comment by funkeleinhorn** [comment]
flashprog.passthru.tests on x86_64-darwin failed to build:

```
cli_common.c:68:86: error: declaration shadows a variable in the global scope [-Werror,-Wshadow]
int cli_parse_log_args(struct log_args *const args, const int opt, const char *const optarg)
```

File: pkgs/by-name/fl/flashprog/package.nix [hunk 1]
```diff
@@ -14,12 +14,12 @@
...
 stdenv.mkDerivation (finalAttrs: {
   pname = "flashprog";
-  version = "1.2";
+  version = "1.3";
 
   src = fetchgit {
     url = "https://review.sourcearcade.org/flashprog";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-Z09hZ4a/G3DhWCmSkPyKs7ecSFBUfez7IWWxIhH3LyI=";
+    rev = "9fc7d7bffa587b1ef82d313e4a098f5afe0a17da";
```
[the remark this answers is not in the document]
Comment (felixsinger) [reply pkgs/by-name/fl/flashprog/package.nix]: Sorry for that. I did some tests on my local branch before and changed the revision to a pre-release one. After the release I've just updated it to the release commit without thinking about it. I changed it back now.

**Comment by felixsinger** [comment]
> flashprog.passthru.tests on x86_64-darwin failed to build:
> 
> ```
> cli_common.c:68:86: error: declaration shadows a variable in the global scope [-Werror,-Wshadow]
> int cli_parse_log_args(struct log_args *const args, const int opt, const char *const optarg)
> ```

I would ignore that issue, and maybe even disable Darwin builds, because there is no way for me to test flashprog on these platforms and I also don't know anyone who could fix the issues.

**Review by funkeleinhorn — APPROVED** [review]
LGTM

**Comment by felixsinger** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by funkeleinhorn** [comment]
> @NixOS/nixpkgs-merge-bot merge

From https://github.com/NixOS/nixpkgs-merge-bot README:
`For now we only allow pull requests done by r-ryantm to be merged`

**Comment by felixsinger** [comment]
Right, but #305350 lists all prs where this command was used and I think people kind of track it.

**Pull request merged**